File: helidon-se/src/main/java/com/github/frederikpietzko/dto/PetDto.java
package com.github.frederikpietzko.dto;

import com.github.frederikpietzko.model.Pet;
import com.github.frederikpietzko.model.Type;


public record PetDto(
  Long id,
  String name,
  int age,
  Type type
) {
  public static PetDto fromEntity(Pet entity) {
    return new PetDto(entity.id(), entity.name(), entity.age(), entity.type());
  }
}

File: javalin/src/main/kotlin/com/github/frederikpietzko/dto/PetDto.kt
package com.github.frederikpietzko.dto

data class PetDto(
  val id: Long,
  val name: String,
  val age: Int,
  val type: String,
)

fun com.github.frederikpietzko.model.Pet.toDto() = PetDto(
  id = id!!,
  name = name,
  age = age,
  type = type.name,
)
